test/unit/test_current_container_finder.py

```diff
@@ -0,0 +1,118 @@
+import dataclasses
+from test.utils.mock_cast import mock_cast
+from typing import (
+    Dict,
+    List,
+    Union,
+)
+from unittest.mock import (
+    MagicMock,
+    PropertyMock,
+    call,
+    create_autospec,
+)
+
+import docker
+import ifaddr
+import pytest
+from docker.models.containers import Container
+
+from exasol.current_container_finder import (
+    CurrentContainerFinder,
+    IPRetriever,
+)
+
+
+def create_ip_retriever_mock(ip_addresses: List[str]) -> Union[MagicMock, IPRetriever]:
+    ip_retriever_mock: Union[MagicMock, IPRetriever] = create_autospec(IPRetriever)
+    ip_mocks = [create_ip_mock(ip_address) for ip_address in ip_addresses]
+    mock_cast(ip_retriever_mock.ips).return_value = ip_mocks
+    return ip_retriever_mock
+
+
+def create_docker_client_mock(
+    containers: List[Dict[str, str]]
+) -> Union[docker.DockerClient, MagicMock]:
+    docker_client_mock: Union[MagicMock, docker.DockerClient] = create_autospec(
+        docker.DockerClient
+    )
+    container_mocks = [create_container_mock(networks) for networks in containers]
+    mock_cast(docker_client_mock.containers.list).return_value = container_mocks
+    return docker_client_mock
+
+
+def create_ip_mock(ip_address: str) -> Union[MagicMock, ifaddr.IP]:
+    ip_mock: Union[MagicMock, ifaddr.IP] = create_autospec(ifaddr.IP)
+    type(ip_mock).ip = PropertyMock(return_value=ip_address)
+    return ip_mock
+
+
+def create_container_mock(networks: Dict[str, str]) -> Union[MagicMock, Container]:
+    container_mock: Union[MagicMock, Container] = create_autospec(Container)
+    type(container_mock).attrs = PropertyMock(
+        return_value={
+            "NetworkSettings": {
+                "Networks": {name: {"IPAddress": ip} for name, ip in networks.items()}
+            }
+        }
+    )
+    return container_mock
+
+
+class TestSetup:
+    def __init__(self, containers: List[Dict[str, str]], ip_addresses: List[str]):
+        self.docker_client_mock = create_docker_client_mock(containers)
+        self.ip_retriever_mock = create_ip_retriever_mock(ip_addresses)
+        self.current_container_finder = CurrentContainerFinder(
+            docker_client=self.docker_client_mock, ip_retriever=self.ip_retriever_mock
+        )
+
+
+def test_ips_in_ips_of_single_container():
```

Comment:
This test was quite hard to understand for me. 

Questions: 
1. For finding a specific container do the ip addresses of all its networks need to be included in the list of searched ip addresses?

```suggestion
def test_single_container():
    """
    Verify finding a single container with all ip addresses exposed by its networks being contained in the given whitelist.
    """
```

Reply:
yes, we don't know which IP  will match, this method is hacky enough that we need to be prepared for everything.